libs/wingsdk/src/target-awscdk/bucket.ts

```diff
@@ -0,0 +1,108 @@
+import { RemovalPolicy } from "aws-cdk-lib";
+import { Effect } from "aws-cdk-lib/aws-iam";
+import {
+  BlockPublicAccess,
+  BucketEncryption,
+  Bucket as S3Bucket,
+} from "aws-cdk-lib/aws-s3";
+import { BucketDeployment, Source } from "aws-cdk-lib/aws-s3-deployment";
+import { Construct } from "constructs";
+import { Function } from "./function";
+import * as cloud from "../cloud";
+import * as core from "../core";
+
+/**
+ * AWS implementation of `cloud.Bucket`.
+ *
+ * @inflight `@winglang/sdk.cloud.IBucketClient`
+ */
+export class Bucket extends cloud.Bucket {
+  private readonly bucket: S3Bucket;
+  private readonly public: boolean;
+
+  constructor(scope: Construct, id: string, props: cloud.BucketProps = {}) {
+    super(scope, id, props);
+
+    this.public = props.public ?? false;
+
+    this.bucket = new S3Bucket(this, "Default", {
+      encryption: BucketEncryption.S3_MANAGED,
+      blockPublicAccess: this.public ? undefined : BlockPublicAccess.BLOCK_ALL,
+      publicReadAccess: this.public ? true : false,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+  }
+
+  public addObject(key: string, body: string): void {
+    new BucketDeployment(this, `S3Object-${key}`, {
+      destinationBucket: this.bucket,
+      sources: [Source.data(key, body)],
+    });
+  }
+
+  /** @internal */
+  public _bind(host: core.IInflightHost, ops: string[]): void {
```

Comment:
I'm thinking that our code for inflights, and anything else that can be reused between AWS targets can go into `libs/wingsdk/src/shared-aws`.

For example, we could create a function `calculateBucketPermissions(arn: string, ops: string[]): PolicyStatements[]` that is shared between both `tf-aws` and `awscdk` to reduce the amount of repetition here.

Reply:
Nice

Reply:
Hey @Chriscbr 

I did what you suggested and I think it was really cool. I just had to be careful with the policies because CDK and TF use slightly different objects (CDK policies has `actions` and `resources` keys and Terraform has `action` and `resource`)